src/core/thread/link_metrics.hpp

```diff
@@ -336,7 +336,11 @@ class Subject : public InstanceLocator, private NonCopyable
 
 private:
     // Max number of SeriesInfo that could be allocated by the pool.
+#if OPENTHREAD_FTD
     static constexpr uint16_t kMaxSeriesSupported = OPENTHREAD_CONFIG_MLE_LINK_METRICS_MAX_SERIES_SUPPORTED;
```

Comment:
How about if we change the default value for `OPENTHREAD_CONFIG_MLE_LINK_METRICS_MAX_SERIES_SUPPORTED` based on `OPENTHREAD_FTD` or `OPENTRHEAD_MTD`?

Reply:
This may cause some warnings and errors when building. I met similar problems before. Because the OT config will be included by some components outside the core or cli, for example, mbedtls, openthread-platform-utils (under example/utils). These components don't have compile options like `OPENTHREAD_FTD` or `OPENTHREAD_MTD` when building. (Because we don't have FTD or MTD version of these libraries) So when evaluating the macros some warnings may be generated and `OPENTHREAD_FTD` or `OPENTHREAD_MTD` will both be `0` since these compile options are not defined when building these the modules. I cannot think of a good solution for this. For now we avoid using `OPENTHREAD_FTD/OPENTHREAD_MTD` in core config files. Otherwise we need to build libs of different versions for them like core and cli.

Reply:
Sorry! Forgot to resolve this. Sounds good to not use the `MTD/FTD` in config header.